### PR TITLE
inference: follow #40985, special case `_typeof_captured_variable`

### PR DESCRIPTION
**Pull request by aviatesk**
/cc @simeonschaub

### CODE REVIEW

File: base/reflection.jl [hunk 1]
```diff
@@ -588,7 +588,7 @@ isdispatchtuple(@nospecialize(t)) = (@_pure_meta; isa(t, DataType) && (t.flags &
 
 iskindtype(@nospecialize t) = (t === DataType || t === UnionAll || t === Union || t === typeof(Bottom))
 isconcretedispatch(@nospecialize t) = isconcretetype(t) && !iskindtype(t)
-using Core: has_free_typevars
+import Core: has_free_typevars
```
Comment (simeonschaub) [inline base/reflection.jl:591]: Any particular reason for this change? `has_free_typevars` should not be overloadable anyways, right?

**Review by simeonschaub — COMMENTED** [review]
This is great! I am happy we don't need a special version of `_typeof_captured_variable` in `Core.Compiler` anymore. I guess the hang @tkf originally reported in #35970 might have just been some kind of bad recursion when bootstrapping?

File: base/reflection.jl (hunk 1 above)
Comment (aviatesk) [reply base/reflection.jl:591]: ah, that's just because I'm almost always using `import` just because of https://github.com/JuliaLang/julia/issues/36529.

Comment (simeonschaub) [reply base/reflection.jl:591]: Ok, makes sense

**Pull request merged**